File: SproInstall/SocketPro/tutorial/VBNet/SampleFour/Client/RAdo.vb
Imports SocketProAdapter
Imports SocketProAdapter.ClientSide
Imports USOCKETLib

Public Class CRAdo
    Inherits CAsyncAdohandler
    Public Sub New()
        MyBase.New(RAdoConst.sidCRAdo)
    End Sub

    Public Sub New(ByVal cs As CClientSocket)
        MyBase.New(RAdoConst.sidCRAdo, cs)
    End Sub

    Public Sub New(ByVal cs As CClientSocket, ByVal DefaultAsyncResultsHandler As IAsyncResultsHandler)
        MyBase.New(RAdoConst.sidCRAdo, cs, DefaultAsyncResultsHandler)
    End Sub

    Public ReadOnly Property CurrentDataTable() As DataTable
        Get
            Return m_AdoSerialier.CurrentDataTable
        End Get
    End Property

    Public Function GetDataSet(ByVal strSQL0 As String, ByVal strSQL2 As String) As DataSet
        Dim bProcessRy As Boolean = ProcessR0(RAdoConst.idGetDataSetCRAdo, strSQL0, strSQL2)
        Return m_AdoSerialier.CurrentDataSet
    End Function

    Public Function GetDataReader(ByVal strSQL As String) As DataTable
        Dim bProcessRy As Boolean = ProcessR0(RAdoConst.idGetDataReaderCRAdo, strSQL)
        Return m_AdoSerialier.CurrentDataTable
    End Function

    Public Function SendDataSet(ByVal ds As DataSet) As Boolean
        Dim b As Boolean
        Send(ds)
        Dim bProcessRy As Boolean = ProcessR1(RAdoConst.idSendDataSetCRAdo, b)
        Return b
    End Function

    Public Function SendDataReader(ByVal dr As IDataReader) As Boolean
        Dim b As Boolean
        Send(dr)
        Dim bProcessRy As Boolean = ProcessR1(RAdoConst.idSendDataReaderCRAdo, b)
        Return b
    End Function
End Class

File: SproInstall/SocketPro/samples/RAdo/CSharp/Client/RAdo.cs

using System;
using SocketProAdapter;
using SocketProAdapter.ClientSide;
using USOCKETLib;
using System.Data;

public class CRAdo : CAsyncAdohandler 
{
    public CRAdo() 
        : base(RAdoConst.sidCRAdo)
    {
    }

    public CRAdo(CClientSocket cs)
        : base(RAdoConst.sidCRAdo, cs)
    {
    }

    public CRAdo(CClientSocket cs, IAsyncResultsHandler DefaultAsyncResultsHandler)
        : base(RAdoConst.sidCRAdo, cs, DefaultAsyncResultsHandler)
    {
    }

	public DataTable CurrentDataTable
	{
		get
		{
			return m_AdoSerialier.CurrentDataTable;
		}
	}

	public DataSet GetDataSet(string strSQL0, string strSQL2)
	{
        bool bProcessRy = ProcessR0(RAdoConst.idGetDataSetCRAdo, strSQL0, strSQL2);
		return m_AdoSerialier.CurrentDataSet;
	}

	public DataTable GetDataReader(string strSQL)
	{
        bool bProcessRy = ProcessR0(RAdoConst.idGetDataReaderCRAdo, strSQL);
		return m_AdoSerialier.CurrentDataTable;
	}

	public bool SendDataSet(DataSet ds)
	{
        bool b;
        Send(ds);
        bool bProcessR1 = ProcessR1(RAdoConst.idSendDataSetCRAdo, out b);
		return b;
	}

	public bool SendDataReader(IDataReader dr)
	{
        bool b;
        Send(dr);
        bool bProcessRy = ProcessR1(RAdoConst.idSendDataReaderCRAdo, out b);
		return b;
	}
}
